notifier/libnotify.go

```diff
@@ -1,28 +1,42 @@
 package notifier
 
 import (
-	"os/exec"
 	"sync"
 
 	log "github.com/sirupsen/logrus"
+	libnotify "github.com/menefotto/go-libnotify"
 )
 
-func notifySend(text string) error {
-	return exec.Command("notify-send", text).Run()
-}
-
 // SetupLibnotifyNotifier configures a notifier to show all touch requests with libnotify
 func SetupLibnotifyNotifier(notifiers *sync.Map) {
 	touch := make(chan Message, 10)
 	notifiers.Store("notifier/libnotify", touch)
 
+	init_success := libnotify.Init("yubikey-touch-detector")
+	defer libnotify.UnInit()
+
+	if !init_success {
+		log.Error("Cannot initialize desktop notifications!")
+		return
+	}
+
+	notification := libnotify.NotificationNew("YubiKey is waiting for a touch", "", "")
+	if notification == nil {
+		log.Error("Cannot create desktop notification!")
+		return
+	}
+
 	for {
 		value := <-touch
 		if value == GPG_ON || value == U2F_ON || value == HMAC_ON {
-			err := notifySend("YubiKey is waiting for a touch")
-			if err != nil {
-				log.Error("Cannot send desktop notification: ", err)
-			}
+			// Error check (!= nil) not possible because menefotto/go-libnotify
+			// uses a custom wrapper instead of builtin 'error'
+			notification.Show()
+		}
+		if value == GPG_OFF || value == U2F_OFF || value == HMAC_OFF {
+			// Error check (!= nil) not possible because menefotto/go-libnotify
+			// uses a custom wrapper instead of builtin 'error'
+			notification.Close()
```

Comment:
I thought about opening an issue about this behaviour, but since the library hasn't been updated for nearly four years, I didn't fancy my chances. And starting a fork of my own would only shift the neglection to a different URL. ;)

Your workaround seems fine, I also don't get an error if the notification is closed manually.

Maybe use the workaround for now, and report the issue in parallel? If it gets fixed, we can then switch to a cleaner solution.

Reply:
Agree 👍

Reply:
→ Workaround in 469a325

Reply:
I've created an issue for the library: https://github.com/menefotto/go-libnotify/issues/1